
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd">
<html lang="en">
 <head>
  <meta content="en-us" http-equiv="Content-Language"/>
  <meta content="text/html; charset=utf-8" http-equiv="Content-Type"/>
  <meta content="no-cache, no-store, must-revalidate" http-equiv="Cache-Control"/>
  <meta content="no-cache" http-equiv="Pragma"/>
  <meta content="0" http-equiv="Expires"/>
  <script async="" src="https://www.googletagmanager.com/gtag/js?id=UA-50117757-1">
  </script>
  <script>
   window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-50117757-1');
  </script>
  <title>
   allpairs_global command
  </title>
  <link href="stylesx.css" rel="stylesheet" type="text/css"/>
  <style type="text/css">
   body.c4 {background-color:#c0c0c0;}
  div.c3 {position:absolute; top:45px; left:20px; width:830px; background-color:#ffffff; border-width:10px; border-style:solid;border-color:white;}
  span.c2 {font-weight: bold}
  div.c1 {position:absolute; top:10px; left:20px; width:850px; height:60px;}
    .TopButtonPara       { color:white; background-color:rgb(50,100,150); border-color:rgb(50,100,150); font-family:Arial, Helvetica, sans-serif; font-weight:normal; font-size:9pt; text-align:center; border-width:4px; border-style:solid; }
    .TopButton           { color:white; }
    a.TopButton:link     { text-decoration:none; }
    a.TopButton:visited  { text-decoration:none; }
    a.TopButton:hover    { color:orange; }

    .NewButtonPara       { color:white; background-color:rgb(50,100,150); border-color:rgb(50,100,150); font-family:Arial, Helvetica, sans-serif; font-weight:normal; font-size:9pt; text-align:center; border-width:4px; border-style:solid; }
    .NewButton           { color:white; }
    a.NewButton:link     { text-decoration:none; }
    a.NewButton:visited  { text-decoration:none; }
    a.NewButton:hover    { color:orange; }

    .SideButtonPara       { color:white; font-family:Arial, Helvetica, sans-serif; font-size:9pt; font-weight:normal; text-align:center; line-height:18px; }
    .SideButton           { color:white; }
    a.SideButton:link     { text-decoration:none; }
    a.SideButton:visited  { text-decoration:none; }
    a.SideButton:hover    { color:orange; }
  </style>
 </head>
 <body style="background-color:#c0c0c0;">
  <div>
   <a href="https://drive5.com/usearch">
    <img alt="USEARCH v12" src="usearch12_banner.jpg" style="position:absolute; top:40px; left:10px; padding:0px; border:0px;"/>
   </a>
  </div>
  <div style="position:absolute; top:115px; left:10px; width:850px; background-color:#ffffff; min-height:500px">
   <div style="position:relative; float:left; background-color:#696969; width:125px; left: 0px; min-height:500px; padding:5px; height: 125px;">
    <div class="SideButtonPara" style="text-align:center; padding-top:5px;">
     <a class="SideButton" href="index.html">
      Docs home
     </a>
     <br/>
     <hr style="border:0; border-bottom: 1px solid white;"/>
     <a class="SideButton" href="cmds.html">
      Commands
     </a>
     <br/>
     <a class="SideButton" href="topics.html">
      Topics
     </a>
     <br/>
     <a class="SideButton" href="citation.html">
      Publications
     </a>
     <br/>
    </div>
   </div>
   <div class="ManText" style="left:20px; position: absolute; left:135px; width:695px; background-color:white; padding:10px">
    <h1>
     uchime3_denovo command
    </h1>
    <span class="ManText">
     <br/>
     Chimera detection using an improved version of the
     <a href="uchime2_algo.html">
      UCHIME2 algorithm
     </a>
     . This command is designed for chimera detection in a set of denoised amplicons. The input sequences must have
     <a href="size_annot.html">
      size=nnn; annotations
     </a>
     giving amplicon abundances. See
     <a href="citation.html">
      UCHIME2 paper
     </a>
     for details.
     <br/>
     <br/>
     The main change from the original UCHIME2 algorithm is that the default minimum abundance skew (abskew option) is now 16 rather than 2. Based on recent results (not yet written up), I believe that with abskew=2 there are many more false positive chimera detections. Perfect chimera detection is not possible due to unbiquitous fake models (see UCHIME2 paper), but with abskew=16 I believe there is a much more reasonable balance between false positives and false negatives.
     <br/>
     <br/>
     Note that the
     <a href="cmd_unoise3.html">
      unoise command
     </a>
     does chimera filtering automatically using exactly the same algorithm as uchime3_denovo, so there is typically no reason to use this command in a USEARCH-based pipeline. It is mainly useful for chimera filtering of amplicons that were denoised by third-party software. In particular, I believe that the DADA2 (at least through v1.4.0) has a high false positive rate for chimera detection, and it would therefore be better to filter chimeras using uchime3_denovo rather than the native DADA2 code.
     <br/>
     <br/>
     The input to uchime3_denovo
     <strong>
      <span class="auto-style1">
       must
      </span>
     </strong>
     <span class="auto-style1">
      be denoised amplicons
     </span>
     . It is
     <strong>
      <span class="auto-style2">
       not
      </span>
     </strong>
     <span class="auto-style2">
      designed to handle noisy reads as input
     </span>
     (even if they have been quality filtered), or to take OTUs as input
     <strong class="ManText">
      .
     </strong>
     <br/>
     <br/>
     The following output files are supported:
     <br/>
     -uchimeout (tabbed text filename)
     <br/>
     -nonchimeras (FASTA file with non-chimeric sequences)
     <br/>
     -chimeras (FASTA file with chimeric sequences)
     <br/>
     -alnout (text file with human-readable alignments)
    </span>
    <p class="ManText">
     <strong class="ManText">
      Chimera detection in an OTU pipeline
      <br/>
     </strong>
     I do
     <strong>
      not recommend
     </strong>
     using uchime2_ref or uchime3_denovo in an OTU clustering pipeline. The
     <a href="cmd_cluster_otus.html">
      cluster_otus command
     </a>
     has built-in de novo chimera filtering which works very well for most data. Using uchime2_ref as a post-processing step is quite likely to discard some false positives that are actually good sequences.
    </p>
    <p>
     <span class="ManText c3">
      Example
     </span>
    </p>
    <p>
     <span class="ManCode">
      usearch -uchime3_denovo denoised.fa -uchimeout out.txt -chimeras ch.fa -nonchimeras nonch.fa
      <br/>
     </span>
    </p>
   </div>
  </div>
 </body>
</html>
