
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd">
<html lang="en">
 <head>
  <meta content="en-us" http-equiv="Content-Language"/>
  <meta content="text/html; charset=utf-8" http-equiv="Content-Type"/>
  <meta content="no-cache, no-store, must-revalidate" http-equiv="Cache-Control"/>
  <meta content="no-cache" http-equiv="Pragma"/>
  <meta content="0" http-equiv="Expires"/>
  <script async="" src="https://www.googletagmanager.com/gtag/js?id=UA-50117757-1">
  </script>
  <script>
   window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-50117757-1');
  </script>
  <title>
   allpairs_global command
  </title>
  <link href="stylesx.css" rel="stylesheet" type="text/css"/>
  <style type="text/css">
   body.c4 {background-color:#c0c0c0;}
  div.c3 {position:absolute; top:45px; left:20px; width:830px; background-color:#ffffff; border-width:10px; border-style:solid;border-color:white;}
  span.c2 {font-weight: bold}
  div.c1 {position:absolute; top:10px; left:20px; width:850px; height:60px;}
    .TopButtonPara       { color:white; background-color:rgb(50,100,150); border-color:rgb(50,100,150); font-family:Arial, Helvetica, sans-serif; font-weight:normal; font-size:9pt; text-align:center; border-width:4px; border-style:solid; }
    .TopButton           { color:white; }
    a.TopButton:link     { text-decoration:none; }
    a.TopButton:visited  { text-decoration:none; }
    a.TopButton:hover    { color:orange; }

    .NewButtonPara       { color:white; background-color:rgb(50,100,150); border-color:rgb(50,100,150); font-family:Arial, Helvetica, sans-serif; font-weight:normal; font-size:9pt; text-align:center; border-width:4px; border-style:solid; }
    .NewButton           { color:white; }
    a.NewButton:link     { text-decoration:none; }
    a.NewButton:visited  { text-decoration:none; }
    a.NewButton:hover    { color:orange; }

    .SideButtonPara       { color:white; font-family:Arial, Helvetica, sans-serif; font-size:9pt; font-weight:normal; text-align:center; line-height:18px; }
    .SideButton           { color:white; }
    a.SideButton:link     { text-decoration:none; }
    a.SideButton:visited  { text-decoration:none; }
    a.SideButton:hover    { color:orange; }
  </style>
 </head>
 <body style="background-color:#c0c0c0;">
  <div>
   <a href="https://drive5.com/usearch">
    <img alt="USEARCH v12" src="usearch12_banner.jpg" style="position:absolute; top:40px; left:10px; padding:0px; border:0px;"/>
   </a>
  </div>
  <div style="position:absolute; top:115px; left:10px; width:850px; background-color:#ffffff; min-height:500px">
   <div style="position:relative; float:left; background-color:#696969; width:125px; left: 0px; min-height:500px; padding:5px; height: 125px;">
    <div class="SideButtonPara" style="text-align:center; padding-top:5px;">
     <a class="SideButton" href="index.html">
      Docs home
     </a>
     <br/>
     <hr style="border:0; border-bottom: 1px solid white;"/>
     <a class="SideButton" href="cmds_all.html">
      Commands
     </a>
     <br/>
     <a class="SideButton" href="topics.html">
      Topics
     </a>
     <br/>
     <a class="SideButton" href="citation.html">
      Publications
     </a>
     <br/>
    </div>
   </div>
   <div class="ManText" style="left:20px; position: absolute; left:135px; width:695px; background-color:white; padding:10px">
    <h1>
     unoise3 command
    </h1>
    <p class="ManText">
     <strong>
      See also
      <br/>
     </strong>
     <a href="uparse_pipeline.html">
      OTU / denoising analysis pipeline
     </a>
     <strong>
      <br/>
     </strong>
     <a href="citation.html">
      UNOISE paper
     </a>
     <br/>
     <a href="faq_uparse_or_unoise.html">
      Should I use UPARSE or UNOISE?
     </a>
    </p>
    <p class="ManText">
     Uses the
     <a href="unoise_algo.html">
      UNOISE algorithm
     </a>
     to perform denoising (error-correction) of amplicon reads.
    </p>
    <p class="ManText">
     Errors are corrected as follows:
     <br/>
     - Reads with sequencing error are identified and corrected.
     <br/>
     - Chimeras are removed.
     <br/>
     <br/>
     Input is a set of quality-filtered unique read sequences with
     <a href="size_annot.html">
      size=nnn; abundance annotations
     </a>
     . See
     <a href="uparse_pipeline.html">
      OTU / denoising pipeline
     </a>
     for details of how reads should be pre-processed and how other types of errors and artifacts can be removed.
    </p>
    <p class="ManText">
     The input file must be sorted by decreasing abundance, i.e. by decreasing value of the size=nnn annotation. The can be done using the
     <a href="DELETE_URL">
      sortbysize command
     </a>
     .
    </p>
    <p class="ManText">
     <span class="ManText">
      The algorithm is designed for Illumina reads, it
      <a href="faq_unoise_not_illumina.html">
       does not work as well on 454, Ion Torrent or PacBio reads
      </a>
      .
     </span>
    </p>
    <p class="ManText">
     Predicted correct biological sequences are written to the -zotus file in FASTA format. Labels are formatted as Zotu
     <em>
      nnn
     </em>
     where
     <em>
      nnn
     </em>
     is 1, 2, 3...
    </p>
    <p class="ManText">
     Predicted correct amplicon sequences are written to the -ampout fle in FASTA format. These include chimeras, so this output file is not generally needed in a production pipeline. Labels are formatted as Amp
     <em>
      nnn
     </em>
     ;uniq=
     <em>
      Uniqlabel
     </em>
     ;uniqsize=
     <em>
      u
     </em>
     ;size=
     <em>
      s
     </em>
     ; where
     <em>
      nnn
     </em>
     is 1, 2, 3...,
     <em>
      Uniqulabel
     </em>
     is the label in the input file, truncated at the first semi-colon,
     <em>
      u
     </em>
     is the size= annotation from the input file and
     <em>
      s
     </em>
     is the total size of reads derived from this amplicon.
    </p>
    <p class="ManText">
     An OTU table can be generated using the
     <a href="cmd_otutab.html">
      otutab command
     </a>
     . See
     <a href="uparse_pipeline.html">
      OTU / denoising pipeline
     </a>
     .
    </p>
    <p class="ManText">
     The -minsize option specifies the minimum abundance (size= annotation). Default is 8. Input sequences with lower abundances are discarded. Most of the low-abundance sequences are usually noisy and are be mapped to a ZOTU by the
     <a href="cmd_otutab.html">
      otutab command
     </a>
     . For higher sensivity, reducing minsize to 4 is reasonable, especially if samples are denoised indivudually rather
     <a href="pool_samples.html">
      pooling all samples together
     </a>
     , as I would usually recommend. With smaller minsize, there tends to be more errors in the predicted low-abundance biological sequences.
    </p>
    <p class="ManText">
     The -tabbedout option specifies a tabbed text filename which reports the processing done for each sequence, e.g. if it is classified as noisy or chimeric.
    </p>
    <p class="ManText">
     The -unoise_alpha option specifies the alpha parameter (see
     <a href="citation.html">
      UNOISE2 paper
     </a>
     for definition). Default is 2.0.
    </p>
    <p>
     <span class="ManText c3">
      Example
     </span>
    </p>
    <p>
     <span class="ManCode">
      usearch -unoise3 uniques.fa -zotus zotus.fa -tabbedout unoise3.txt
      <br/>
     </span>
    </p>
   </div>
  </div>
 </body>
</html>
