
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd">
<html lang="en">
 <head>
  <meta content="en-us" http-equiv="Content-Language"/>
  <meta content="text/html; charset=utf-8" http-equiv="Content-Type"/>
  <meta content="no-cache, no-store, must-revalidate" http-equiv="Cache-Control"/>
  <meta content="no-cache" http-equiv="Pragma"/>
  <meta content="0" http-equiv="Expires"/>
  <script async="" src="https://www.googletagmanager.com/gtag/js?id=UA-50117757-1">
  </script>
  <script>
   window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-50117757-1');
  </script>
  <title>
   allpairs_global command
  </title>
  <link href="stylesx.css" rel="stylesheet" type="text/css"/>
  <style type="text/css">
   body.c4 {background-color:#c0c0c0;}
  div.c3 {position:absolute; top:45px; left:20px; width:830px; background-color:#ffffff; border-width:10px; border-style:solid;border-color:white;}
  span.c2 {font-weight: bold}
  div.c1 {position:absolute; top:10px; left:20px; width:850px; height:60px;}
    .TopButtonPara       { color:white; background-color:rgb(50,100,150); border-color:rgb(50,100,150); font-family:Arial, Helvetica, sans-serif; font-weight:normal; font-size:9pt; text-align:center; border-width:4px; border-style:solid; }
    .TopButton           { color:white; }
    a.TopButton:link     { text-decoration:none; }
    a.TopButton:visited  { text-decoration:none; }
    a.TopButton:hover    { color:orange; }

    .NewButtonPara       { color:white; background-color:rgb(50,100,150); border-color:rgb(50,100,150); font-family:Arial, Helvetica, sans-serif; font-weight:normal; font-size:9pt; text-align:center; border-width:4px; border-style:solid; }
    .NewButton           { color:white; }
    a.NewButton:link     { text-decoration:none; }
    a.NewButton:visited  { text-decoration:none; }
    a.NewButton:hover    { color:orange; }

    .SideButtonPara       { color:white; font-family:Arial, Helvetica, sans-serif; font-size:9pt; font-weight:normal; text-align:center; line-height:18px; }
    .SideButton           { color:white; }
    a.SideButton:link     { text-decoration:none; }
    a.SideButton:visited  { text-decoration:none; }
    a.SideButton:hover    { color:orange; }
  </style>
 </head>
 <body style="background-color:#c0c0c0;">
  <div>
   <a href="https://drive5.com/usearch">
    <img alt="USEARCH v12" src="usearch12_banner.jpg" style="position:absolute; top:40px; left:10px; padding:0px; border:0px;"/>
   </a>
  </div>
  <div style="position:absolute; top:115px; left:10px; width:850px; background-color:#ffffff; min-height:500px">
   <div style="position:relative; float:left; background-color:#696969; width:125px; left: 0px; min-height:500px; padding:5px; height: 125px;">
    <div class="SideButtonPara" style="text-align:center; padding-top:5px;">
     <a class="SideButton" href="index.html">
      Docs home
     </a>
     <br/>
     <hr style="border:0; border-bottom: 1px solid white;"/>
     <a class="SideButton" href="cmds.html">
      Commands
     </a>
     <br/>
     <a class="SideButton" href="topics.html">
      Topics
     </a>
     <br/>
     <a class="SideButton" href="citation.html">
      Publications
     </a>
     <br/>
    </div>
   </div>
   <div class="ManText" style="left:20px; position: absolute; left:135px; width:695px; background-color:white; padding:10px">
    <h1>
     UNOISE algorithm
    </h1>
    <p class="ManText">
     <strong>
      See also
      <br/>
     </strong>
     <a href="citation.html">
      UNOISE paper
     </a>
     <br/>
     <a href="faq_uparse_or_unoise.html">
      Should I use UPARSE (97% OTUs) or UNOISE (denoising)?
     </a>
    </p>
    <p class="ManText">
     The UNOISE algorithm performs error-correction (
     <a href="denoising.html">
      denoising
     </a>
     ) on amplicon reads. It is implemented in the
     <a href="cmd_unoise3.html">
      unoise3 command
     </a>
     .
    </p>
    <p class="ManText">
     The original UNOISE algorithm was briefly described in
     <a href="https://doi.org/10.1093/bioinformatics/btv401">
      Edgar &amp; Flyvbjerg (2015)
     </a>
     . An improved algorithm was described and validated in
     <a href="https://doi.org/10.1101/081257">
      Edgar 2016
     </a>
     . The implementation in
     <a href="cmd_unoise3.html">
      unoise3
     </a>
     and
     <a href="cmd_uchime3_denovo.html">
      uchime3_denovo
     </a>
     is quite similar to UNOISE2 except for a change in parameters for chimera detection, which I believe greatly reduces the number of false positives over the original parameters described in the UNOISE2 paper that were  implemented in the earlier unoise2 and uchime2_denovo commands in usearch v9.
    </p>
    <p class="ManText">
     <span class="ManText">
      The algorithm is designed for Illumina reads, it
      <a href="faq_unoise_not_illumina.html">
       does not work as well on 454, Ion Torrent or PacBio reads
      </a>
      .
     </span>
    </p>
    <p class="ManText">
     Correct biological sequences are recovered from the reads, resolving distinct sequences down to a single difference (often) or two or more differences (almost always).
    </p>
    <p class="ManText">
     Errors are corrected as follows:
     <br/>
     - Reads with sequencing and PCR point error are identified and removed.
     <br/>
     - Chimeras are removed.
    </p>
    <p class="ManText">
     Abundances are calculated after denoising by generating an
     <a href="otu_table.html">
      OTU table
     </a>
     using the
     <a href="cmd_otutab.html">
      otutab command
     </a>
     .
    </p>
    <p class="ManText">
     <img alt="Image" src="unoise2.gif"/>
    </p>
    <p class="ManText">
     <strong>
      Schematic of the UNOISE2 denoising strategy (figure from the UNOISE2 paper).
     </strong>
     <br/>
     The left panel shows the neighborhood close to a high-abundance unique read sequence X, grouped by the number of sequence differences (d). Dots are unique sequences, the size of a dot indicates its abundance. Green dots are correct biological sequences; red dots have one or more errors. Neighbors with small numbers of differences and small abundance compared to X are predicted to be bad reads of X. The right panel shows the denoised amplicons. Here, X and b were correctly predicted, e is an error with anomalously high abundance that was wrongly predicted to be correct, f is an error that was correctly discarded but has an abundance almost high enough to be a false positive, and g is a low-abundance correct amplicon that was wrongly discarded. The abundances of b, e, and f are similar, illustrating the fundamental challenge in denoising: how to set an abundance threshold that distinguishes correct sequences from errors.
     <br/>
    </p>
   </div>
  </div>
 </body>
</html>
